tests/test_app.py

```diff
@@ -1,4 +1,228 @@
+    i = 1
+    test_item = Item(name='Harry',status='Done',dateLastActivity=today_str)
+    while i <= 4:
+        test_list.append(test_item)
+        i += 1
+    return test_list      
+
+@pytest.fixture
+def board_five_in_done_today():
+    today = datetime.now()
+    today_str = datetime.strftime(today,'%Y-%m-%dT%H:%M:%S.%fZ')
+    test_list=[]
+    i = 1
+    test_item = Item(name='Harry',status='Done',dateLastActivity=today_str)
+    while i <= 5:
+        test_list.append(test_item)
+        i += 1
+    return test_list   
+
+@pytest.fixture
+def board_done_1today_2yesterday_3last_week_4tomorrow():
+    test_list=[]
+    today = datetime.now()
+    tomorrow = today + timedelta(days=1)
+    yesterday = today - timedelta(days=1)
+    lastweek = today - timedelta(days=7)
+    today_str = datetime.strftime(today,'%Y-%m-%dT%H:%M:%S.%fZ')
+    yesterday_str = datetime.strftime(yesterday,'%Y-%m-%dT%H:%M:%S.%fZ')
+    lastweek_str = datetime.strftime(lastweek,'%Y-%m-%dT%H:%M:%S.%fZ')
+    tomorrow_str = datetime.strftime(tomorrow,'%Y-%m-%dT%H:%M:%S.%fZ')
+
+    test_item = Item(name='Tom is done today', dateLastActivity = today_str, status='Done')
+    test_list.append(test_item)
+    test_item = Item(name='Dick is done yesterday', dateLastActivity = yesterday_str, status='Done')
+    test_list.append(test_item)
+    test_list.append(test_item)
+    test_item = Item(name='Harry is done last week',dateLastActivity = lastweek_str, status='Done')
+    test_list.append(test_item)
+    test_list.append(test_item)
+    test_list.append(test_item)
+    test_item = Item(name='Nobody can be done tomorrow',dateLastActivity = tomorrow_str, status='Done')
+    test_list.append(test_item)
+    test_list.append(test_item)
+    test_list.append(test_item)
+    test_list.append(test_item)
+
+    return test_list    
+
+@pytest.fixture
+def board_one_in_each():
+    tom_item = Item(name='Tom',status='To Do')
+    dick_item = Item(name='Dick',status='Doing')
+    harry_item = Item(name='Harry',status='Done')
+    test_list=[]
+    test_list.append(tom_item)
+    test_list.append(dick_item)
+    test_list.append(harry_item)
+    return test_list    
+
+@pytest.fixture
+def board_one_hundred_in_each():
+    tom_item = Item(name='Tom',status='To Do')
+    dick_item = Item(name='Dick',status='Doing')
+    harry_item = Item(name='Harry',status='Done')
+    test_list=[]
+    i = 1
+    while i <= 100:
+        test_list.append(tom_item)
+        test_list.append(dick_item)
+        test_list.append(harry_item)
+        i += 1
+    return test_list    
+
+def test_no_items():
+    test_model=ViewModel([])
+    assert len(test_model.to_do_items) == 0
+    assert len(test_model.doing_items) == 0
+    assert len(test_model.done_items) == 0
+
+def test_one_in_to_do(board_one_in_to_do):
+    test_model=ViewModel(board_one_in_to_do)
+    assert len(test_model.to_do_items) == 1
+    assert len(test_model.doing_items) == 0
+    assert len(test_model.done_items) == 0
+ 
```

Comment:
One nice potential addition could be to check that the status of the item comes back correctly - e.g.:
```
assert test_model.to_do_items[0].status == 'To Do'
```
This is pretty minor here, but e.g. in the one-of-each test it might be more helpful to check that the right items have been returned in the right properties

Reply:
https://github.com/ianm1968/ToDoTake2/commit/28f9c50523e515c4bf75f6e19603edf1f7c87a9a